genprotocol/kbgitkbfs-avdl/disk_block_cache.avdl

```diff
@@ -0,0 +1,55 @@
+// Copyright 2017 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+/**
+  DiskBlockCacheInterface specifies how to access a disk cache remotely.
+  */
+@namespace("kbgitkbfs.1")
+protocol DiskBlockCache {
+
+  import idl "github.com/keybase/client/go/protocol/keybase1" as keybase1;
+
+  enum PrefetchStatus {
+      NO_PREFETCH_0,
+      TRIGGERED_PREFETCH_1,
+      FINISHED_PREFETCH_2
+  }
+
+  /**
+    GetCachedBlockRes is the response from GetBlock.
+    */
+  record GetBlockRes {
+    bytes buf;
+    string serverHalf;
+    PrefetchStatus prefetchStatus;
+  }
+
+  /**
+    DeleteBlocksRes is the response from DeleteBlocks.
+    */
+  record DeleteBlocksRes {
+    int numRemoved;
+    long sizeRemoved;
+  }
+
+  /**
+    GetBlock gets a block from the disk cache.
+    */
+  GetBlockRes GetBlock(keybase1.TLFID tlfID, string blockID);
```

Comment:
I modeled it off the `BServerRemote` protocol, which uses a string.

Reply:
Also: I think strings are just `[]byte` under the hood: https://blog.golang.org/strings#TOC_2.

> As far as the content of a string is concerned, it is exactly equivalent to a slice of bytes.

Reply:
Right but with string, the data has to be decoded from hex: https://github.com/keybase/kbfs/blob/0738ce3bf5e2c01ee1d172b7eff98f179ccdf382/kbfshash/hash.go#L121-L139

Anyway, thought it might be nice to do something more efficient here.  Not sure if it was a mistake to do it that way on bserver, since it makes the remote protocol more debuggable.  But here in a local protocol, might be better to go with what's most efficient.  I won't push on it though.

Reply:
Good call, I'll change it, thanks!

Reply:
TLF ID too, I guess.